subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AbstractAuthenticationSupportedRepository.java

```diff
@@ -77,6 +78,11 @@ public void credentials(Action<? super PasswordCredentials> action) {
         delegate.credentials(credentialsType, action);
     }
 
+    public void credentials(Provider<Credentials> credentials) {
```

Comment:
If we would like to promote the `repositories { credentials(provider) }` syntax, this method needs to be added to `org.gradle.api.artifacts.repositories.AuthenticationSupported` and overridden here.